src/main/scala/codechicken/multipart/block/TileMultiPart.java

```diff
@@ -341,13 +344,17 @@ public void setRemoved() {
     //endregion
 
     //region *** Internal callbacks ***
-    public VoxelShape getOutlineShape() { return outlineShapeHolder.update(partList); }
+    public VoxelShape getOutlineShape(ISelectionContext context) { return outlineShapeHolder.update(partList, part -> part.getOutlineShape(context)); }
+
+    public VoxelShape getCollisionShape(ISelectionContext context) { return collisionShapeHolder.update(partList, part -> part.getCollisionShape(context)); }
+
+    public VoxelShape getRenderOcclusionShape() { return occlusionShapeHolder.update(partList, TMultiPart::getRenderOcclusionShape); }
 
-    public VoxelShape getCollisionShape() { return collisionShapeHolder.update(partList); }
+    public VoxelShape getInteractionShape() { return interactionShapeHolder.update(partList, TMultiPart::getInteractionShape); }
 
-    public VoxelShape getRenderOcclusionShape() { return occlusionShapeHolder.update(partList); }
+    public VoxelShape getSupportShape() { return supportShapeHolder.update(partList, TMultiPart::getSupportShape); }
 
-    public VoxelShape getInteractionShape() { return interactionShapeHolder.update(partList); }
+    public VoxelShape getVisualShape(ISelectionContext context) { return visualShapeHolder.update(partList, part -> part.getVisualShape(context)); }
```

Comment:
These are missing CCL changes, we should discuss how to handle caching of these with ISelectionContext's properly.